pallets/parachain-staking/src/lib.rs

```diff
@@ -175,35 +195,221 @@ pub mod pallet {
 		}
 		pub fn bond_more(&mut self, more: B) {
 			self.bond += more;
-			self.total += more;
+			self.total_counted += more;
+			self.total_backing += more;
 		}
 		// Return None if less >= self.bond => collator must leave instead of bond less
 		pub fn bond_less(&mut self, less: B) -> Option<B> {
 			if self.bond > less {
 				self.bond -= less;
-				self.total -= less;
+				self.total_counted -= less;
+				self.total_backing -= less;
 				Some(self.bond)
 			} else {
 				None
 			}
 		}
-		pub fn inc_nominator(&mut self, nominator: A, more: B) {
-			for x in &mut self.nominators.0 {
+		/// Infallible sorted insertion
+		/// caller must verify !self.nominators.contains(nominator.owner) before call
+		pub fn add_top_nominator(&mut self, nominator: Bond<A, B>) {
+			match self
+				.top_nominators
+				.binary_search_by(|x| nominator.amount.cmp(&x.amount))
+			{
+				Ok(i) => self.top_nominators.insert(i, nominator),
+				Err(i) => self.top_nominators.insert(i, nominator),
+			}
+		}
+		/// Infallible sorted insertion
+		/// caller must verify !self.nominators.contains(nominator.owner) before call
+		pub fn add_bottom_nominator(&mut self, nominator: Bond<A, B>) {
+			match self
+				.bottom_nominators
+				.binary_search_by(|x| x.amount.cmp(&nominator.amount))
+			{
+				Ok(i) => self.bottom_nominators.insert(i, nominator),
+				Err(i) => self.bottom_nominators.insert(i, nominator),
+			}
+		}
+		/// Sort top nominators from greatest to least
+		pub fn sort_top_nominators(&mut self) {
+			self.top_nominators
+				.sort_unstable_by(|a, b| b.amount.partial_cmp(&a.amount).unwrap());
+		}
+		/// Sort bottom nominators from least to greatest
+		pub fn sort_bottom_nominators(&mut self) {
+			self.bottom_nominators
+				.sort_unstable_by(|a, b| a.amount.partial_cmp(&b.amount).unwrap());
```

Comment:
You've used the trait bound `B: Ord`, so you should be able to call [`cmp`](https://doc.rust-lang.org/std/cmp/trait.Ord.html#tymethod.cmp) rather than `partial_cmp`. Then you won't have to unwrap. (If I understood correctly)